src/main/java/io/confluent/connect/elasticsearch/ElasticsearchWriter.java

```diff
@@ -253,6 +253,8 @@ public void write(Collection<SinkRecord> records) {
       final boolean ignoreSchema =
           ignoreSchemaTopics.contains(sinkRecord.topic()) || this.ignoreSchema;
 
+      client.createIndices(Collections.singleton(index));
```

Comment:
just saw this one. I think this violates the `on.start` in the documentation.

Reply:
I don't think so, this makes sure that indices are created during the write operations. Imagine the situation where you are using a regexp SMT to rewrite the topicToIndexName, this indices would not be created at start time.

Actually this made me realise the position of this call is wrong, it should always be done after the mapping is created, otherwise elasticsearch will create them "automatically".

Happy to collaborate with you.

Reply:
To extend this, I'm proposing this flow with this PR.

1. By default: create always an initial topic to index during start time. This could be deactivated if necessary.
2. Make sure that if a SMT is applied, the index is created anyway. (this does not happen in the current way of operation).

Reply:
Thanks @purbon for describing the motivation, but here are a couple of things to consider:

1. This call forces an index creation request to ES for every message consumed by every sink task. That's going to be expensive, and will reduce connector throughput.
2. If we are going to create indices per record, then we don't really ever need to create them at bootstrap time. One of those is redundant. 

IMHO, for this PR, we should punt the index-creates-at-write-time to a future PR since that might need deeper changes (or follow up with PR https://github.com/confluentinc/kafka-connect-elasticsearch/pull/261).

Reply:
Hi @wicknicks,
  speaking about 1, yes this is inevitable, it will be an extra over head, the only thing we can do here is to manage our internal index name cache, as proposed in #261, however this will as well include more extra challenges.  
  i included 2 as a backwards compatible feature. we can decide to don't do it, but this was my intention.

Reply:
This is good now. We have a cache that keeps track of created indices.

Reply:
Good, happy to be of help.